sway-parse/src/expr/mod.rs

```diff
@@ -555,7 +559,11 @@ fn parse_projection(parser: &mut Parser, ctx: ParseExprCtx) -> ParseResult<Expr>
                 };
                 continue;
             }
-            return Err(parser.emit_error(ParseErrorKind::ExpectedFieldName));
+
+            // Nothing expected followed. Now we have parsed `expr .`.
+            // Try to recover as an unknown sort of expression.
+            parser.emit_error(ParseErrorKind::ExpectedFieldName);
+            return Ok(Expr::Error(Span::join(target.span(), dot_token.span())));
```

Comment:
I wonder if instead of joining the span's here, if we could instead return a vector of them? This would make accessing the unique parts of the string more ergonomic on the lsp side.